core/modules/dashboard/dashboard.module

```diff
@@ -0,0 +1,356 @@
+      'restrict access' => TRUE,
+    ),
+  );
+}
+
+/**
+ * Implements hook_config_info().
+ */
+function dashboard_config_info() {
+  $prefixes['layout.layout.dashboard'] = array(
+    'label' => t('Dasboard default layout'),
+    'group' => t('Configuration'),
+  );
+  $prefixes['dashboard.settings'] = array(
+    'label' => t('Dasboard settings'),
+    'group' => t('Configuration'),
+  );
+  return $prefixes;
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function dashboard_block_info() {
+  $blocks = array();
+
+  // Dashboard blocks.
+  $blocks['create'] = array(
+    'info' => t('Dashboard - Create content'),
+    'description' => t('Provides links to create new content'),
+    'class' => 'DashboardCreateBlock',
+  );
+  $blocks['menu'] = array(
+    'info' => t('Dashboard - Menus'),
+    'description' => t('Provides links to administer menus'),
+    'class' => 'DashboardMenuBlock',
+  );
+  $blocks['node_types'] = array(
+    'info' => t('Dashboard - Content types'),
+    'description' => t('Provides links to administer content types'),
+    'class' => 'DashboardContentTypesBlock',
+  );
+  $blocks['overview_content'] = array(
+    'info' => t('Dashboard - Content'),
+    'description' => t('Displays a summary of content statistics'),
+    'class' => 'DashboardOverviewContentBlock',
+  );
+  $blocks['overview_user'] = array(
+    'info' => t('Dashboard - User accounts'),
+    'description' => t('Displays a summary of user account statistics'),
+    'class' => 'DashboardOverviewUserBlock',
+  );
+  $blocks['taxonomy'] = array(
+    'info' => t('Dashboard - Categories'),
+    'description' => t('Provides links to administer taxonomy'),
+    'class' => 'DashboardTaxonomyBlock',
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_autoload_info().
+ */
+function dashboard_autoload_info() {
+  return array(
+    'DashboardCreateBlock' => 'dashboard_blocks/block.create.inc',
+    'DashboardMenuBlock' => 'dashboard_blocks/block.menus.inc',
+    'DashboardContentTypesBlock' => 'dashboard_blocks/block.node_types.inc',
+    'DashboardOverviewContentBlock' => 'dashboard_blocks/block.overview_content.inc',
+    'DashboardOverviewUserBlock' => 'dashboard_blocks/block.overview_user.inc',
+    'DashboardTaxonomyBlock' => 'dashboard_blocks/block.taxonomy.inc',
+  );
+}
+
+/**
+ * Implements hook_user_login().
+ * Redirects to the dashboard if configured.
+ */
+function dashboard_user_login(&$edit, $account) {
+  // Don't redirect for password reset.
+  if (!((arg(0) == 'user') && (arg(1) == 'reset'))) {
+    // Check for permission to use dashboard.
+    if (user_access('access dashboard', $account)) {
+      $dashboard = layout_load('dashboard');
+      if ($dashboard && !$dashboard->disabled && $dashboard->checkAccess()) {
+        // Check for redirect to dashboard setting.
+        if (config_get('dashboard.settings', 'dashboard_login_redirect') == 1) {
+          $_GET['destination'] = 'admin/dashboard';
```

Comment:
Is this the same way that Login Redirect modules work? Seems a bit overpowered versus using a form_alter(). We should also at least check that we're not overriding an existing `destination`. e.g. if filling out a form at `user/login?destination=node/add` you should still be taken to the `node/add`, not the dashboard.

Reply:
I've had a few looks at the other redirect modules and the internet. Setting `$_GET['destination']` seems a common practice. [Login destination module](https://cgit.drupalcode.org/login_destination/tree/login_destination.module#n158) mentions they use a form alter to allow external URLs, but we don't need that. 
I agree we should account for existing `$_GET['destination']s`.